core/src/main/java/feign/Contract.java

```diff
@@ -132,6 +134,10 @@ protected MethodMetadata parseAndValidateMetadata(Class<?> targetType, Method me
     private static void checkMapString(String name, Class<?> type, Type genericType) {
       checkState(Map.class.isAssignableFrom(type),
               "%s parameter must be a Map: %s", name, type);
+      checkMapKeys(name, genericType);
+    }
+
+    private static void checkMapKeys(String name, Type genericType) {
```

Comment:
this looks like a good add - but I didn't see a test covering it in the case that it isn't a string (I assume tests already cover the case where it is)

Reply:
This wasn't an addition, it was a DRY split. I suppose I could revert the `checkMapString` method and just 2x leave the `Map` instance check even though I just checked that. The test is in `DefaultContractTest.queryMapKeysMustBeStrings`.